scripts/powershell/Validate-PipelineYaml.ps1

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/env pwsh
```

Comment:
The general idea looks great. However given the nature of this script I'd suggest putting it in https://dev.azure.com/azure-sdk/internal/_git/azure-sdk-devops which is more about doing this higher level analysis on our specific devops pipelines.